gstreamer1: enable build options necessary for most applications

flyn-org opened the pull request:
Commit cb058bf changed gstreamer1's build options, and this broke some
applications.

First, -Doption-parsing=false is intended for very specific embedded
applications rather general packages. Please see:

https://github.com/openwrt/packages/pull/8847
https://gitlab.freedesktop.org/gstreamer/gstreamer/-/issues/388

Second, -Dregistry=false seems to break plugin discovery. Symptoms of
this include broken applications and gst-inspect-1.0 listing zero plugins.

Fixes #13531.

Signed-off-by: W. Michael Petullo <mike@flyn.org>

Maintainer: me
Description:

neheb commented:
Please bump PKG_RELEASE.

flyn-org commented:
Bumped PKG_RELEASE.

hnyman commented:
Is this intentionally against 19.07 (instead of master)?

If this is a cherry-pick from master , please mention it in the commit message

flyn-org commented:
Yes, GStreamer in the stable branch is broken. See https://github.com/openwrt/packages/issues/13531. We fixed this in master in August with commit 10aa094d7f7b039b04eec3feab9e6c4a6a5756a7, but we have not until now fixed this in stable. Thankfully, @arkanoid87 reported the issue there.

I pushed a new commit message that references the cherry pick in the style of commit eee7ce2b4963e2e89e19d1a35aef285079554a27.
